.github/workflows/main.yml

```diff
@@ -0,0 +1,23 @@
+name: Main Workflow
+
+on: [push]
+
+jobs:
+  build:
+    name: Test Docker Image
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Test with pytest
+        run: |
+          pytest
```

Comment:
This doesn't work for a compact repository, need to configure workflow with docker-compose or docker image respectively to run services like db worker etc